src/workerd/util/sqlite.c++

```diff
@@ -492,44 +519,47 @@ bool SqliteDatabase::isAuthorized(int actionCode,
     case SQLITE_CREATE_VIEW        :   /* View Name       NULL            */
     case SQLITE_DROP_VIEW          :   /* View Name       NULL            */
     case SQLITE_REINDEX            :   /* Index Name      NULL            */
+      VALUE_GATE(param3, nullptr, "main"_kj);
       KJ_ASSERT(param2 == nullptr);
       return regulator.isAllowedName(KJ_ASSERT_NONNULL(param1));
 
     case SQLITE_ALTER_TABLE        :   /* Database Name   Table Name      */
-      // Why is the database name passed redundantly here? Weird.
-      KJ_ASSERT(KJ_ASSERT_NONNULL(param1) == "main"_kj);
+      VALUE_GATE(param1, "main"_kj); // NOTE, expect dbName on param1 here
       return regulator.isAllowedName(KJ_ASSERT_NONNULL(param2));
 
     case SQLITE_READ               :   /* Table Name      Column Name     */
     case SQLITE_UPDATE             :   /* Table Name      Column Name     */
+      VALUE_GATE(param3, nullptr, "main"_kj, "temp"_kj);
       return regulator.isAllowedName(KJ_ASSERT_NONNULL(param1));
 
     case SQLITE_CREATE_INDEX       :   /* Index Name      Table Name      */
     case SQLITE_DROP_INDEX         :   /* Index Name      Table Name      */
     case SQLITE_CREATE_TRIGGER     :   /* Trigger Name    Table Name      */
     case SQLITE_DROP_TRIGGER       :   /* Trigger Name    Table Name      */
+      VALUE_GATE(param3, nullptr, "main"_kj);
       return regulator.isAllowedName(KJ_ASSERT_NONNULL(param1)) &&
              regulator.isAllowedName(KJ_ASSERT_NONNULL(param2));
 
     case SQLITE_TRANSACTION        :   /* Operation       NULL            */
+      VALUE_GATE(param3, nullptr, "main"_kj);
       {
         // Verify param1 is one of the values we expect.
-        kj::StringPtr op = KJ_ASSERT_NONNULL(param1);
-        KJ_ASSERT(op == "BEGIN" || op == "ROLLBACK" || op == "COMMIT", op);
```

Comment:
I really did want this to be an assert, an not just return false if the op is not one of these. The purpose of the assert is to check my assumption that these are the _only_ valid values -- if this isn't true, then the code has a bug, and I want this to be flagged in Sentry.